controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/ArtifactExpirer.java

```diff
@@ -73,4 +92,35 @@ private boolean isExpired(Artifact artifact, Instant now, VersionStatus versionS
         return true;
     }
 
+    /** Model versions in use in this system, and, if this is a CD system, in the main/public system */
+    private Set<Version> modelVersionsInUse() {
+        var system = controller().system();
+        var versions = versionsForSystem(system);
+
+        if (controller().system().isCd()) {
```

Comment:
This `if` is unnecessary as it is covered by the below `if (system == SystemName.PublicCd)` etc